tests/cephfs/xfs_test.py

```diff
@@ -0,0 +1,157 @@
+            "git clone https://github.com/markfasheh/ocfs2-tools.git",
+            "cd ocfs2-tools && ./autogen.sh && ./configure && make && make install && cd ..",
+            "yum install -y libacl-devel bc libtool",
+            "git clone git://git.kernel.org/pub/scm/fs/xfs/xfstests-dev.git",
+            "cd xfstests-dev && make && sudo make install",
+        ]
+        log.info("Installing required packages for xfstests")
+        for command in commands:
+            client1.exec_command(sudo=True, cmd=command, check_ec=False)
+        rand = "".join(
+            random.choice(string.ascii_lowercase + string.digits) for _ in range(5)
+        )
+        # Define mount directories
+        log.info("Mounting only kernel clients for xfstests")
+        test_mount_point = f"/mnt/cephfs_kernel_{rand}_test"
+        scratch_mount_point = f"/mnt/cephfs_kernel_{rand}_scratch"
+        # Mount CephFS
+        mon_node_ips = fs_util.get_mon_node_ips()
+        fs_util.kernel_mount([client1], test_mount_point, ",".join(mon_node_ips))
+        fs_util.kernel_mount([client1], scratch_mount_point, ",".join(mon_node_ips))
+        # create subdirectory in the mounted directory
+        client1.exec_command(sudo=True, cmd=f"mkdir {test_mount_point}/aa")
+        client1.exec_command(sudo=True, cmd=f"mkdir {scratch_mount_point}/bb")
+        # get admin key
+        admin_key, _ = client1.exec_command(
+            sudo=True, cmd="ceph auth get-key client.admin"
+        )
+        xfs_config_context = f"""
+        export TEST_DIR={test_mount_point}
+        export SCRATCH_MNT={scratch_mount_point}
+        export FSTYP=ceph
+        export TEST_DEV={mon_node_ips[0]}:/aa
+        export SCRATCH_DEV={mon_node_ips[1]}:/bb
+
+        COMMON_OPTIONS="name=admin"
+        COMMON_OPTIONS+=",secret={admin_key}"
+
+        TEST_FS_MOUNT_OPTS="-o ${{COMMON_OPTIONS}}"
+        MOUNT_OPTIONS="-o ${{COMMON_OPTIONS}}"
+
+        export TEST_FS_MOUNT_OPTS
+        export MOUNT_OPTIONS
+        """
+        # write to file as local.config.
+        client1.exec_command(
+            sudo=True,
+            cmd=f"echo '{xfs_config_context}' > /root/xfstests-dev/local.config",
+        )
+        # create exclude file
+        exclude_file = "generic/003"
+        client1.exec_command(
+            sudo=True, cmd=f"echo '{exclude_file}' > /root/xfstests-dev/ceph.exclude"
+        )
+        # run the tests
+        client1.exec_command(
+            sudo=True,
+            cmd="cd /root/xfstests-dev && ./check -d -T -g quick -e ceph.exclude",
+            check_ec=False,
+            long_running=True,
+        )
+        log.info("XFS tests completed successfully")
+        return 0
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        uninstall_commands = [
+            "dnf remove -y acl attr automake bc dbench dump e2fsprogs fio gawk gcc gdbm-devel git"
```

Comment:
Hope amar's comment on bash script fopr series of pkg add can also be leveraged here for pkg remove?

Reply:
yes. it can be. what would be difference between creating a 2 bash files and having cmd lines in list..?